### PR TITLE
Migrate from Teamcity to Github actions

### PR DESCRIPTION
**Pull request by davidfurey**
## What does this change?

This change adopt GitHub Actions for CI, replacing TeamCity which DevX are switching off in Q2 2023.

This change isn't exactly a no-op switch, as it switches from deprecated node-riffraff-artifact plugin to devx recommendation guardian/actions-riff-raff.

The localstack setup is based on https://docs.localstack.cloud/user-guide/ci/github-actions/. These docs say that "An official GitHub action for this also planned, to make the configuration easier and less verbose.". If this is ever released, would could simplify our config by switching to it.

## How to test
- [x] Deploy to CODE

## Notes
- I've paused the [TeamCity project](https://teamcity.gutools.co.uk/buildConfiguration/EditorialTools_UserTelemetryService?mode=builds).
- `.zip` filesizes are different between build 700 and build 701, but unzipping the deb yields no differences so I presume this is a quirk of the zip compression

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      - main
+jobs:
+  CI:
+    runs-on: ubuntu-latest
+    permissions:
+      # required by aws-actions/configure-aws-credentials
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v3
```
Comment (samanthagottlieb) [inline .github/workflows/ci.yml:21]: There's a v4 of this action

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+      - uses: aws-actions/configure-aws-credentials@v2
```
Comment (samanthagottlieb) [inline .github/workflows/ci.yml:23]: Also a v4 of this

**Review by samanthagottlieb — APPROVED** [review]
A couple of comments about the latest versions of actions, but this otherwise looks good to me!

**Comment by davidfurey** [comment]
Deployed, data still seems to be coming through in PROD

**Pull request merged**